pkgs/by-name/xk/xkeysnail/package.nix

```
@@ -1,38 +1,38 @@
 {
   lib,
   fetchFromGitHub,
-  makeWrapper,
   python3Packages,
+  fetchpatch,
 }:
 
 python3Packages.buildPythonApplication rec {
   pname = "xkeysnail";
-  version = "0.4.0";
+  version = "0.4";
 
   src = fetchFromGitHub {
     owner = "mooz";
     repo = pname;
     rev = "bf3c93b4fe6efd42893db4e6588e5ef1c4909cfb";
-    sha256 = "0plcpb4ndzfsd5hj32m0g32swnhyph9sd759cdhhzmjvlq3j8q6p";
+    hash = "sha256-12AkB6Zb1g9hY6mcphO8HlquxXigiiFhadr9Zsm6jF4=";
   };
 
-  nativeBuildInputs = [ makeWrapper ];
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/mooz/xkeysnail/pull/181.patch";
```

The contents of this path can change if the linked PR is updated, breaking this derivation. 

There's a similar issue where a URL like `https://github.com/mooz/xkeysnail/pull/181/commits/457ab424fb32c4bfc6e6ea307752a2ce5d77853b.patch` can start to 404 if the PR is force-pushed.

Therefore, you should link to the PR in a comment and use a link directly to the relevant commit:

```suggestion
      # https://github.com/mooz/xkeysnail/pull/181
      url = "https://github.com/shicks/xkeysnail/commit/457ab424fb32c4bfc6e6ea307752a2ce5d77853b.patch";
```

---

That's also true, but considerably less likely in my experience. Fortunately(?), all forks on GitHub share a backing repository, so we could refer to the commit on the fork from the upstream repository instead:

```
https://github.com/mooz/xkeysnail/commit/457ab424fb32c4bfc6e6ea307752a2ce5d77853b.patch
```